tests/stream_test.ts

```diff
@@ -0,0 +1,810 @@
+
+    await Promise.all(
+      [
+        client.xadd(key, 1000, { "field": "foo" }),
+        client.xadd(key, 2000, { "field": "bar" }),
+      ],
+    );
+
+    const initialConsumer = "someone";
+
+    const firstReply = await client.xreadgroup(
+      [{ key, xid: ">" }],
+      { group, consumer: initialConsumer },
+    );
+
+    const firstPending = await client.xpending(key, group);
+
+    await sleepMillis(5);
+
+    assertEquals(firstPending.count, 2);
+    assertNotEquals(firstPending.startId, firstPending.endId);
+    assertEquals(firstPending.consumers.length, 1);
+    assertEquals(firstPending.consumers[0].name, "someone");
+    assertEquals(firstPending.consumers[0].pending, 2);
+
+    const minIdleTime = 4;
+
+    // minimum options
+    const claimingConsumer = "responsible-process";
+    const firstClaimed = await client.xclaim(
+      key,
+      { group, consumer: claimingConsumer, minIdleTime },
+      1000,
+      2000,
+    );
+    switch (firstClaimed.kind) {
+      case "messages":
+        assertEquals(firstClaimed.messages.length, 2);
+        assertEquals(
+          firstClaimed.messages[0].field_values,
+          new Map(Object.entries({ "field": "foo" })),
+        );
+        assertEquals(
+          firstClaimed.messages[1].field_values,
+          new Map(Object.entries({ "field": "bar" })),
+        );
+        break;
+      default:
+        assert(false);
+    }
+
+    // ACK these messages so we can try XPENDING/XCLAIM
+    // on a new batch
+    await client.xack(key, group, ...firstReply[0].messages.map((m) => m.xid));
+
+    // Let's write some more messages and try
+    // other formats of XPENDING/XCLAIM
+    await Promise.all(
+      [
+        client.xadd(key, 3000, { "field": "foo" }),
+        client.xadd(key, [3000, 1], { "field": "bar" }),
+        client.xadd(key, [3000, 2], { "field": "baz" }),
+      ],
+    );
+
+    const secondReply = await client.xreadgroup(
+      [{ key, xid: ">" }],
+      { group, consumer: initialConsumer },
+    );
+
+    // take a short nap and increase the lastDeliveredMs
+    await sleepMillis(5);
+
+    // try another form of xpending: counts for all consumers (we have only one)
+    const secondPending = await client.xpending_count(
+      key,
+      group,
+      { start: "-", end: "+", count: 10 },
+    );
+    assertEquals(secondPending.length, 3);
+    for (const info of secondPending) {
+      assertEquals(info.owner, "someone");
+      assert(info.lastDeliveredMs > 4);
+      // We called XREADGROUP so it was delivered once
+      // (but not acknowledged yet!)
+      assertEquals(info.timesDelivered, 1);
+    }
+
+    // the output for justIDs will have a different shape
+    const secondClaimedXIds = await client.xclaim(
+      key,
+      { group, consumer: claimingConsumer, minIdleTime, justXId: true },
+      [3000, 0],
+      [3000, 1],
+      [3000, 2],
+    );
+
+    switch (secondClaimedXIds.kind) {
```

Comment:
Can this switch statement be replaced with the following code?
```suggestion
    assert(firstClaimed.kind === "justxid");
    assertEquals(
      secondClaimedXIds.xids,
      [
        { unixMs: 3000, seqNo: 0 },
        { unixMs: 3000, seqNo: 1 },
        { unixMs: 3000, seqNo: 2 },
      ],
    );
...
```

Reply:
Yes, this improves it !